src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -4785,6 +4785,251 @@ public static void ToLowerInvariant(string s, string expected)
             Assert.Equal(expected, destination.ToString());
         }
 
+        [Fact]
+        public static void ZeroLengthToLower()
+        {
+            char[] expectedSource = { 'a', 'B', 'c' };
+            char[] a = { 'a', 'B', 'c' };
+            var expectedDestination = new char[1] { 'a' };
+            var expectedDestinationString = new char[0];
+            Span<char> destination = new char[1] { 'a' };
+
+            string s1 = new string(a, 2, 0);            
+            Assert.Equal(expectedDestinationString, s1.ToLowerInvariant().ToArray());            
```

Comment:
nit: How about just compare to an empty string and maybe rename `s1` to `emptyString`?
```C#
Assert.Equal("", s1.ToLowerInvariant());
```